File: app/src/main/java/com/example/lukas/trainerapp/db/dao/UserDao.java
package com.example.lukas.trainerapp.db.dao;

import com.example.lukas.trainerapp.db.entity.User;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import io.reactivex.Flowable;

@Dao
public interface UserDao {

    @Query("SELECT * FROM User")
    List<User> getAll();

    @Query("SELECT * FROM User WHERE userId IN (:userIds)")
    List<User> loadAllByIds(long[] userIds);

    @Query("SELECT * FROM User WHERE fullName LIKE :fullName LIMIT 1")
    User findByFullName(String fullName);

    @Query("SELECT * FROM User ORDER BY createdAt DESC LIMIT 1")
    LiveData<User> getUser();

    @Query("SELECT * FROM User ORDER BY createdAt DESC LIMIT 1")
    User getSimpleUser();

    @Insert
    void insertAll(User... users);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertUser(User login);

    @Query("DELETE FROM User")
    void deleteAllUsers();

    @Delete
    void delete(User user);
}

File: app/src/main/java/com/trainerapp/db/dao/UserDao.kt
package com.trainerapp.db.dao

import com.trainerapp.db.entity.User

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.RoomDatabase
import com.facebook.internal.Mutable
import io.reactivex.Flowable

@Dao
interface UserDao {

    @get:Query("SELECT * FROM User")
    val all: List<User>

    @get:Query("SELECT * FROM User ORDER BY createdAt DESC LIMIT 1")
    val user: LiveData<User>

    @get:Query("SELECT * FROM User ORDER BY createdAt DESC LIMIT 1")
    val simpleUser: User

    @Query("SELECT * FROM User WHERE userId IN (:userIds)")
    fun loadAllByIds(userIds: LongArray): List<User>

    @Query("SELECT * FROM User WHERE fullName LIKE :fullName LIMIT 1")
    fun findByFullName(fullName: String): User

    @Insert
    fun insertAll(vararg users: User)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertUser(login: User): Long

    @Query("DELETE FROM User")
    fun deleteAllUsers()

    @Delete
    fun delete(user: User)
}
